### PR TITLE
doc: improve documentation.

### PR DESCRIPTION
**Pull request by iocanel**

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -2,41 +2,56 @@
 
 [![CircleCI](https://circleci.com/gh/ap4k/ap4k.svg?style=svg)](https://circleci.com/gh/ap4k/ap4k)
 
-Annotation processor for Kubernetes is a collection of Java annotation processors and tools for generating Kubernetes/Openshift resources at compile time.
+Ap4k is a collection of Java annotations and processors for generating Kubernetes/Openshift manifests at compile time.
```
Comment (cmoulliard) [inline readme.md]: Why do you use `manifest` as word instead of `resource` which is the exact technical term ?

Comment (iocanel) [reply readme.md]: imho manifest strongly suggests that we are referring to the text representation if the resource. Resource feels a little bit more abstract.

Comment (cmoulliard) [reply readme.md]: If you want to keep this word, then I propose to add a few definitions top of the document to explain what is a manifest, resources, ... Make sense @iocanel @metacosm @geoand

Comment (geoand) [reply readme.md]: To tell you the truth, I consider the two words interchangeable, but indeed semantically `manifest` is more appropriate

Comment (iocanel) [reply readme.md]: @cmoulliard: I prefer using the term that needs less explaining. If this term is resources, then I'll start using that.

wdyt?

Comment (cmoulliard) [reply readme.md]: You can use it but then explain what is a manifest and what is generated

**Comment by iocanel** [comment]
@geoand: as promised added some docs on external manifest support. also add tons of doc on examples.

**Pull request merged**